stdlib/public/core/Sort.swift.gyb

```diff
@@ -364,9 +364,272 @@ public func sorted<
 ) -> [C.Generator.Element] {
   fatalError("unavailable function can't be called")
 }
+
+
+/// Performs a binary search in `from`..<`to` for the value at `val`,
+/// preserving the lower ordering of `val` in case of equal matches.
+///
+/// - Requires: The values in `from`..<`to` must already be in order.
+/// - Returns: The stable insertion point for the value in `val`.
+private func _stableSearchWithLower<
+  T${"" if p else ": Comparable"}
+>(
+  from: UnsafeMutablePointer<T>,
+  _ to: UnsafeMutablePointer<T>,
+  _ val: UnsafeMutablePointer<T> ${"," if p else ""}
```

Comment:
Revamped these to use `start` and `end` throughout for consistency with Swift's terminology.